src/components/CreateAccount/CreateAccount.tsx

```diff
@@ -1,8 +1,325 @@
+    // eslint-disable-next-line no-lonely-if
+    if (isAccountAvailable === null) {
+      accountStatusMessage = 'Checking availability...';
+    } else if (isAccountAvailable) {
+      accountStatusMessage = `${formValues?.username}.${network.fastAuth.accountIdSuffix} is available!`;
+      accountStatusState = 'success';
+    } else {
+      accountStatusMessage = `${formValues?.username}.${network.fastAuth.accountIdSuffix} is taken, try something else.`;
+      accountStatusState = 'error';
+    }
+  }
+
   return (
-    <div>CreateAccount route</div>
+    <StyledContainer>
+      <FormContainer onSubmit={onSubmit}>
+        <header>
+          <h1>Create account</h1>
+          <p className="desc">Use this account to sign in everywhere on NEAR, no password required.</p>
+        </header>
+
+        <InputContainer>
+          <label htmlFor="email">Email</label>
+
+          <input
+            {...register('email', {
+              required: 'Please enter a valid email address',
+              pattern:  {
+                value:   emailPattern,
+                message: 'Please enter a valid email address',
+              },
+            })}
+            onChange={(e) => {
+              clearErrors('email');
+              setValue('email', e.target.value);
+              if (!isValidEmail(e.target.value)) return;
+              if (!formValues?.username || !touchedFields?.username) {
+                setValue('username', getEmailId(e.target.value));
+              }
+            }}
+            placeholder="user_name@email.com"
+            type="email"
+          />
+          {/* shouldn't need to do a type check here but message is not resolving as a string for some reason */}
+          {typeof errors.email?.message === 'string' && <ErrorText role="alert">{errors.email?.message}</ErrorText>}
+        </InputContainer>
+
+        <InputContainer>
+          <label htmlFor="username">Account ID</label>
+          <input
+            autoComplete="webauthn username"
+            {...register('username', {
+              required: 'Please enter a valid account ID',
+              pattern:  {
+                value:   accountAddressPatternNoSubaccount,
+                message: 'Please enter a valid account ID',
+              },
+              validate: () => {
+                if (!isAccountAvailable) {
+                  return 'Please enter a valid account ID';
+                }
+                return null;
+              },
+            })}
+            placeholder="user_name.near"
+          />
+          <p className="subText">
+            <span className={accountStatusState || ''}>{accountStatusMessage}</span>
+          </p>
+          {/* shouldn't need to do a type check here but message is not resolving as a string for some reason */}
+          {typeof errors.username?.message === 'string' && (
+            <ErrorText role="alert">{errors.username?.message}</ErrorText>
+          )}
```

Comment:
Should probably be using ErrorMessage component here or displaying the appropriate accountStatusMessage here programmatically based on checking form state directly once `validate` has been configured correctly
https://react-hook-form.com/docs/useformstate/errormessage